### PR TITLE
Make CMake 3.10 the minimum version, add LTO option

### PR DESCRIPTION
**Pull request by romainguy**

### CODE REVIEW

File: build/windows/ci-common.bat [hunk 1]
```diff
@@ -1,13 +1,21 @@
...
+	choco install llvm --version 8.0.0 -y
 
-    :: refreshenv is necessary to put LLVM on path
+    :: retry if choco install failed
+    if errorlevel 1 (
+        choco install llvm --version 8.0.8 -y
+        if errorlevel 1 exit /b %errorlevel%
+    )
+
+    :: refreshenv is necessary to put Cm=Make and LLVM on path
```
Comment (bejado) [inline build/windows/ci-common.bat]:
```suggestion
    :: refreshenv is necessary to put CMake and LLVM on path
```

File: build/windows/ci-common.bat [hunk 2]
```diff
@@ -1,13 +1,21 @@
...
+    choco install cmake -y
 
     :: retry if choco install failed
     if errorlevel 1 (
-        choco install llvm --version 6.0.1 -y
+        choco install cmake -y
         if errorlevel 1 exit /b %errorlevel%
     )
+	
+	choco install llvm --version 8.0.0 -y
```
Comment (bejado) [inline build/windows/ci-common.bat]: nit: spacing

File: CMakeLists.txt [hunk 3]
```diff
@@ -124,6 +126,20 @@ if ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang" AND "${CMAKE_CXX_SIMULATE_ID}" S
     set(CLANG_CL true)
 endif()
 
+# ==================================================================================================
+# Link time optimizations (LTO)
+# ==================================================================================================
+if (ENABLE_LTO)
+	include(CheckIPOSupported)
```
Comment (bejado) [inline CMakeLists.txt]: nit: I think you're using tabs here, not spaces

File: build/windows/ci-common.bat (hunk 2 above)
Comment (romainguy) [reply build/windows/ci-common.bat]: Just realized my TextMate is set to use Tabs… :/

**Comment by bejado** [comment]
Why do we now need to explicitly install `CMake` on Windows kokoro?

File: build/windows/ci-common.bat (hunk 2 above)
Comment (bejado) [reply build/windows/ci-common.bat]: 😮

**Comment by romainguy** [comment]
I switched to requiring CMake 3.10 minimum and our Windows CI has CMake 3.7 (the GCP Windows Server 2016 variant has CMake 3.10 though… go figure).

File: CMakeLists.txt [hunk 4]
```diff
@@ -101,7 +103,7 @@ set(TOOLS ${CMAKE_CURRENT_SOURCE_DIR}/tools)
 # ==================================================================================================
 # Compiler check
 # ==================================================================================================
-set(MIN_CLANG_VERSION "5.0")
+set(MIN_CLANG_VERSION "7.0")
```
Comment (romainguy) [inline CMakeLists.txt]:
```suggestion
set(MIN_CLANG_VERSION "6.0")
```

File: android/filament-android/CMakeLists.txt [hunk 5]
```diff
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 3.4.1)
+cmake_minimum_required(VERSION 3.10)3.4.1)
```
Comment (romainguy) [inline android/filament-android/CMakeLists.txt]:
```suggestion
cmake_minimum_required(VERSION 3.10)
```

File: android/gltfio-android/CMakeLists.txt [hunk 6]
```diff
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 3.4.1)
+cmake_minimum_required(VERSION 3.10)3.4.1)
```
Comment (romainguy) [inline android/gltfio-android/CMakeLists.txt]:
```suggestion
cmake_minimum_required(VERSION 3.10)
```

File: build/windows/ci-common.bat [hunk 7]
```diff
@@ -1,13 +1,21 @@
...
-    choco install llvm --version 6.0.1 -y
+    choco install cmake -y
 
     :: retry if choco install failed
     if errorlevel 1 (
-        choco install llvm --version 6.0.1 -y
+        choco install cmake -y
         if errorlevel 1 exit /b %errorlevel%
     )
+
```
Comment (romainguy) [inline build/windows/ci-common.bat]:
```suggestion

    refreshenv

```

**Pull request merged**